File: annotationprocessor/src/test/resources/Pokemon.java
import com.cr.o.cdc.annotations.GraphQlRequest;
import com.cr.o.cdc.annotations.Input;

@GraphQlRequest(url = "url", name = "pokemon", nullable = true, inputs = {@Input(name = "name", type = String.class)})
public class Pokemon {
    private String name;

    Pokemon(String name) {
        this.name = name;
    }
}

File: app/src/main/java/com/cr/o/cdc/sandboxAndroid/model/Pokemon.kt
package com.cr.o.cdc.sandboxAndroid.model

import com.cr.o.cdc.annotations.GraphQlRequest
import com.cr.o.cdc.requestsannotations.Input
import com.cr.o.cdc.sandboxAndroid.BuildConfig

/**
 * Created by Camilo on 01/01/20.
 */

@GraphQlRequest(BuildConfig.URL_SERVER, "pokemon", [Input("name", String::class)], true)
data class Pokemon(
    val id: String,
    val name: String,
    val weight: PokemonDimension,
    val evolutions: List<PokemonMini>
)
